projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-manager.service.ts

```diff
@@ -64,23 +65,48 @@ export class DataLayerManagerService {
    * You can observe the retrieved layers using one of the manager's Observable properties:
    * [allLayers$], [selectedLayers$], or [availableLayers$].
    */
-  retrieveAll() {
+  retrieveAll(sortCompareFn: LayerCompareFn = () => 0, selectAll: boolean = false) {
     this.reset();
     this.loading$.next(true);
     merge(...this.dataLayerServices.map((service) => service.retrieve()))
       .pipe(takeUntil(this.cancel$))
       .subscribe({
-        next: (layer) =>
-          this.stateSubject.next({
-            ...this.stateSubject.value,
-            layers: this.mergeService.merge(this.state.layers, layer),
-          }),
+        next: (layer) => {
+          const layers = this.mergeService.merge(this.state.layers, layer);
+          let nextState = { ...this.stateSubject.value, layers };
+          if (selectAll) {
+            nextState = this.selectAllLayers(nextState);
+          }
+          nextState = this.sortLayers(nextState, sortCompareFn);
+          this.stateSubject.next(nextState);
+        },
         error: (err) => console.error(err),
         complete: () => {
           this.loading$.next(false);
         },
       });
   }
+  
+  /** Reducer that returns a new state with selected layers sorted by `sortCompareFn` */
+  private sortLayers = produce<DataLayerManagerState, [LayerCompareFn]>((draft, sortCompareFn) => {
+    draft.selected.sort((idA, idB) => sortCompareFn(draft.layers[idA], draft.layers[idB]));
+  });
+
+  /** Reducer that returns a new state with all layers selected */
+  private selectAllLayers = (state: DataLayerManagerState) => {
+    return Object.keys(this.state.layers).reduce((nextState, id) => this.selectLayer(nextState, id), state);
```

Comment:
this should be using layers from the state parameter, not the instance property
```suggestion
    return Object.keys(state.layers).reduce((nextState, id) => this.selectLayer(nextState, id), state);
```